### PR TITLE
Make NavMesh recomputation for non-default agent params optional

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

`simulator.py` `reconfigure()` automatically recomputes the NavMesh if `default_agent` has non-default parameters. Generally this makes sense because most of our saved NavMeshes are generated for the default agent. However, users may want to skip this if they've generated their own custom NavMeshes as in ReplicaCAD dataset for Hab2.0 rearrangement tasks.

~This PR primarily adds a new config option `recompute_navmesh_on_agent_params` which defaults to previous behavior, but if toggled off allows skipping automatic recomputation.~
Edit: This PR adds cache, query, and serialization of navmesh settings accompanying any programmatically computed or loaded NavMesh. Existing NavMeshes computed before this PR are assumed to use default settings. If agent params are different from cached settings for the currently loaded NavMesh, it will be recomputed to match current agent settings. 

In addition, I cleaned up documentation for SimulatorConfiguration and removed two unused legacy fields:
 - `defaultCameraUuid`
 - `compressTextures`

## How Has This Been Tested

CI passes, new tests for NavMesh settings cache and comparators. Built the docs locally.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/esp/sim/SimulatorConfiguration.h [hunk 1]
```diff
@@ -14,19 +14,24 @@ namespace esp {
...
   std::string activeSceneName;
+  //! The default agent id used during initialization and functionally whenever
+  //! alternative agent ids are not provided.
   int defaultAgentId = 0;
+  //! The system GPU device to use for rendering.
   int gpuDeviceId = 0;
+  //! The Simulator and Pathfinder random seed. Set during scene initialization.
   unsigned int randomSeed = 0;
-  std::string defaultCameraUuid = "rgba_camera";
-  bool compressTextures = false;
```
Comment (aclegg3) [inline src/esp/sim/SimulatorConfiguration.h:25]: If these do have a purpose, I couldn't find it.

**Comment by erikwijmans** [comment]
IMO if we are adding something, we should just save the navmesh settings into the navmesh itself and then check against those. The navmesh is saved with a version so we should be able to save this with backwards compatibility and we just use the default navmesh settings.

**Review by Skylion007 — APPROVED** [review]

**Comment by Skylion007** [comment]
I agree with @erikwijmans , we should save the agent size with the navmesh. It seems like an integral property of the navmesh and the only reason we aren't doing it is technical debt.

**Comment by erikwijmans** [comment]
@aclegg3 @Skylion007 I pushed a change set to save the params into the navmesh.

File: src/esp/nav/PathFinder.cpp [hunk 2]
```diff
@@ -37,6 +37,25 @@ namespace Cr = Corrade;
 namespace esp {
 namespace nav {
 
+bool operator==(const NavMeshSettings& a, const NavMeshSettings& b) {
+#define CLOSE(name) ((a.name - b.name) < 1e-5)
```
[the remark this answers is not in the document]
Comment (erikwijmans) [reply src/esp/nav/PathFinder.cpp:41]: Whoops, thank you. Coding is hard

**Review by Skylion007 — APPROVED** [review]

**Review by aclegg3 — COMMENTED** [review]
Looks good, thanks for the upgrade @erikwijmans.

**Pull request merged**